### PR TITLE
[Kernel/Xam] XNotifyGetNext - Check for null in id_ptr

### PR DESCRIPTION
**Pull request by Gliniak**
Really small change to XNotifyGetNext based on _Carcassonne_ game code.

image

PS. Game still gonna crash due to returning error code in _XUsbcamCreate_, but that will be in another PR, as I still don't know whole story 🦊

### CODE REVIEW

File: src/xenia/kernel/xam/xam_notify.cc [hunk 1]
```diff
@@ -74,7 +74,10 @@ dword_result_t XNotifyGetNext(dword_t handle, dword_t match_id,
       *param_ptr = param;
     }
   } else {
-    *id_ptr = 0;
+    // id_ptr may be null - "Carcassonne" explicitly passes nullptr in code
+    if (id_ptr) {
```
Comment (Triang3l) [inline src/xenia/kernel/xam/xam_notify.cc:78]: Not handled in the `if (dequeued)` above.

**Review by Triang3l — CHANGES_REQUESTED** [review]

**Comment by Triang3l** [comment]
Also should the first two early-outs in that function write anything to id_ptr and param_ptr?

File: src/xenia/kernel/xam/xam_notify.cc (hunk 1 above)
Comment (Gliniak) [reply src/xenia/kernel/xam/xam_notify.cc:78]: I don't know if it should be handled in that case too. Maybe it should, but we have no case of such behaviour

Comment (Triang3l) [reply src/xenia/kernel/xam/xam_notify.cc:78]: If it's a nullable parameter in any case ("maybenull" in SAL terms), null pointer should be passable in all cases, I think.

**Comment by Triang3l** [comment]
Was told by @Gliniak in PMs that it returns E_INVALIDARG in the official kernel if either of those arguments is null, so probably need to recheck all the return values of the functions.

**Comment by Gliniak** [comment]
After spending some time I figured out that:
E_INVALIDARG (X_ERROR_INVALID_PARAMETER) - is returned only if id_ptr is null
E_UNEXPECTED is returned only when data under handle is corrupted in some specific way, but not handle itself

**Pull request merged**